From b9ab720e33748cd022f095620e75ca7eba24a965 Mon Sep 17 00:00:00 2001
From: Thomas Egerer <hakke_007@gmx.de>
Date: Tue, 14 Apr 2009 16:15:17 -0700
Subject: [PATCH 14/23] Fix display of xfrm

When using iproute2 to display information on policies installed in kernel
(ip x p s) output is incorrect: IPv6 addresses printed as IPv4 addresses.
In case I am dealing with inter protocol policies where the template's address
family differs from those of the policy itself.
The patch attached solves this problem.
---
 ip/ipxfrm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ip/ipxfrm.c b/ip/ipxfrm.c
index 24b2114..708ce9f 100644
--- a/ip/ipxfrm.c
+++ b/ip/ipxfrm.c
@@ -609,7 +609,7 @@ static void xfrm_tmpl_print(struct xfrm_user_tmpl *tmpls, int len,
 			fputs(prefix, fp);
 
 		xfrm_id_info_print(&tmpl->saddr, &tmpl->id, tmpl->mode,
-				   tmpl->reqid, family, 0, fp, prefix, "tmpl ");
+				   tmpl->reqid, tmpl->family, 0, fp, prefix, "tmpl ");
 
 		if (show_stats > 0 || tmpl->optional) {
 			if (prefix)
-- 
1.6.3.3

